testing/jormungandr-integration-tests/src/common/network/controller.rs

```diff
@@ -39,6 +41,7 @@ pub struct Controller {
     working_directory: TempDir,
     block0_file: PathBuf,
     block0_hash: HeaderId,
+    node_key_files: HashMap<String, NamedTempFile>,
```

Comment:
Is there a special need for these files to exist outside the temp directory controlled by the `working_directory` member? If not, I think you could create them under the `working_directory` hierarchy and map the names as `PathBuf` values.

It will be simpler to keep all the files in stock if we decide we want to preserve the working directory on test failure. Now you'd also need to un-temp all the key files.